draco/synthesis/noise.py

```diff
@@ -332,3 +332,60 @@ def process(self, data_exp):
                 )
 
         return data_exp
+
+
+class GaussianNoiseExternal(task.SingleTask, random.RandomTask):
```

Comment:
It seems like this could be done by using the existing tasks `GaussianNoise` to generate a realisation and then `MixData` to combine them, rather than adding an extra task?!

Reply:
yeah I guess so, I hadn't thought of using `MixData`